modules/apps/dataset/dataset-api/src/main/java/com/liferay/dataset/view/DatasetViewSerializer.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dataset.view;
+
+import com.liferay.portal.kernel.json.JSONArray;
+
+import java.util.Locale;
+
+/**
+ * @author Marco Leo
+ */
+public interface DatasetViewSerializer {
```

Comment:
I'm afraid about this one clashing with what @dsanz is working on which might also be called "View(s)"... @dsanz, do you think that'll be the name of persisted views or do we have a better name for it?

Reply:
This is the thing I was talking about when I said "mixing data and GUI" in Slack. I put all the raw data stuff in the `data` subpackage, leaving the rest for GUI stuff. However that leads to:

1) duplicity (in this case)
2) some blur, because we have, for example, `data` and `filter` or `action` at the same level.

Maybe we can do one more package level to differentiate between the GUI component and its backend :thinking: . Maybe the problem is that we are using the `-api` bundle for two different things... I don't think we have any other place in portal where an API serves a GUI component + a backend service :thinking:

Reply:
:thinking: This is one of those classes that are in the API but are only used inside the taglib so I guess we will be able to move it to the `-taglib` module in a future PR to make it hidden.

Reply:
This serializer seems to serialize a thing called "DatasetDisplayView" which implements the dataset display "mid section" (top→management toolbar, mid→view, bottom→pagination) so we have classes for this implementing the List, Table and Card views. Also, it looks like specific classes exist for particular use cases such as "AllSkusClayTableDataSetDisplayView" amogst others, they seem to be spetializations of the Table view.

In contrast, I've been using "Custom View" in the "custom view management" feature spec jargon. Here, the concept of custom view is more like a set of settings that dictate how dataset display will render. That includes, amongst other, the kind of "Dataset display view" (table, cards,...) plus many other things like ordering of items, filtering, etc. So, we might keep "Custom View" to denote "visualization settings that are saved by end user", whereas "DatasetView" would be an API to implement specific "mid" sections of the dataset display

Does this make sense?

Reply:
There's also the `datasetDisplayName` (like in [here](https://github.com/liferay-frontend/liferay-portal/pull/953/files#diff-ee67ec26f81601cb1fef4b28c88a5de29cf9aac360247c9ecdc159f453504f3cR27)) which I'm not sure how is related to this.

If I understood it correctly a dataset display is the whole thing the user defines so that this framework shows something and the view is a part of it. So, a display would be a screen/page, and a view would be the screen when in list mode, another view would be the screen in a table mode, and so on.

I'm not sure if my display is the same as your custom view...

Reply:
![Screen_Shot_2021-04-06_at_15_06_08](https://user-images.githubusercontent.com/905006/113724269-17c17480-96ea-11eb-9d26-43bedbbf1cff.png)

A CustomView is basically a configuration defining how the DatasetDisplay is configured (filters, pagination, sorting, displayStyle...).

@dsanz, some Figmas here and epics of what's coming might be useful 😉

Reply:
Maybe it would be a good idea to add the `-taglib` module to this PR (or remove more files associated to it from the PR as I did in the [7th commit](https://github.com/liferay-frontend/liferay-portal/pull/953/commits/a877fc3da74d4cf063d5a2d32cdf9cd8599d5471)) because some files (like the `DatasetViewSerializer`) are really an API of the `-taglib` exclusively...

Reply:
We can explore this a bit further to see how it looks... the problem with removing is that then you have to replace all usages which is something we might want to avoid and follow a more traditional deprecation process instead. Consider that these APIs are already released and maybe even in use by customers :/

Reply:
This :point_up: may not be as simple, because some of the things used for the taglib may be needed by the pure JS component too. I'm going to check the code again an read @dsanz 's document.

Reply:
`datasetDisplayName` contains a value which is used to identify different osgi components together via component property "clay.data.set.display.name=", so that implementors can use that value when registering a DatasetDisplayView but also when creating filter classes which will be used in that specific DatasetDisplayView. All these looks part of the taglib machinery, which entry point is the `id` attribute. 

For example, `commerce-product-definitions-web` uses `<clay:headless-data-set-display>` and passes an [specific id](https://github.com/liferay/liferay-portal/blob/1567ae20025a06b58a8c28569afc11f2193f4816/modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/view.jsp#L42) to it, which is used by the taglib to [serialize the view settings, the view itself, and the filters](https://github.com/liferay/liferay-portal/blob/da32e6c5dc6206754f45c93cdffc0cc0d4232eaf/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/HeadlessDataSetDisplayTag.java#L393-L414). The two latter utilize that id to find the proper component implementing the [view specification](https://github.com/liferay/liferay-portal/blob/6db1295ed2ab7bbf5d05a27108a80c399369c249/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/CommerceProductDefinitionClayTableDataSetDisplayView.java#L33) and filters specification (there is one per available filter crieria, like [this](https://github.com/liferay/liferay-portal/blob/4a842e7d7091c61b04eb35a74e28aa5980b4f9c9/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/taglib/clay/data/set/filter/CommerceChannelClayTableDataSetFilter.java#L28)), both things are strongly related as filters only make sense for specific data visualization method.

Custom view could be seen as an enhancement of the view settings serialization, which in the end, provides settings to customize things like visible columns, items order, etc. on top of a pre-existing dataset instance which comes with its own views + filters.